### PR TITLE
Updated the game-tutorial to avoid problems with C#

### PR DESCRIPTION
**Pull request by Husky110**
I've updated the tutorial a bit due to some stumbling through it.
1.) Eventho C# supports the usage of var, it's highly disencouraged due to performance. Especially people new to coding should learn it the clean way.
2.) The functionnames in the connections had to be replaced in C#, otherwise the project will fail and throw method-not-found-exceptions on every connection.

### CODE REVIEW

**Comment by Calinou** [comment]
> 1.) Eventho C# supports the usage of var, it's highly disencouraged due to performance. Especially people new to coding should learn it the clean way.

Can you link to benchmarks pointing to slowdowns when using `var`? I can't see why type inference would slow down code execution.

**Comment by Husky110** [comment]
I gotta say that I just learned it that way. So after reading a bit into it you guys might be right. But, according to this: [https://stackoverflow.com/questions/356846/will-using-var-affect-performance](https://stackoverflow.com/questions/356846/will-using-var-affect-performance) it might have an impact on certain situations. Eventho I still would not recommend it's usage outside of LINQ for visual reasons. See this one: [https://www.intertech.com/Blog/the-use-and-abuse-of-the-c-var-keyword/](https://www.intertech.com/Blog/the-use-and-abuse-of-the-c-var-keyword/)

**Comment by Husky110** [comment]
To be fair - the var-issue might have been an personal thing and I don't wanna start a code-religion-issue here. ;) 
So eventho if 1.) does not apply, I still see a reason to change the other things highlighted in the commit. Like the change for the "retart the editor"-issue and the functionname-issue.

File: getting_started/step_by_step/your_first_game.rst [hunk 1]
```diff
@@ -795,15 +798,15 @@ Add the following code:
...
         mobSpawnLocation.SetOffset(rand.Next());
 
         // Create a Mob instance and add it to the scene.
-        var mobInstance = (RigidBody2D) Mob.Instance();
+        RigidBody2D mobInstance = (RigidBody2D) Mob.Instance();
         AddChild(mobInstance);
 
         // Set the mob's direction perpendicular to the path direction.
-        var direction = mobSpawnLocation.Rotation + Mathf.Pi / 2;
+        float direction = mobSpawnLocation.Rotation + Mathf.Pi / 2;
```
[the remark this answers is not in the document]
Comment (neikeq) [reply getting_started/step_by_step/your_first_game.rst]: Yes, this one change is right. However, I noticed something wrong in that guide. I recommended not using `var` in for loops. e.g.: `int i = 0` vs `var i = 0`, prefer the former.

**Comment by Warlaan** [comment]
I recommend that you close this PR and create a new one with just the non-var-related changes.

Regarding var: in my experience as a teacher there are good reasons to use var (not in all but in many situations) and the arguments against it are usually based on either a conservative gut feeling or in misunderstandings of code architecture and design.

If you are interested I could explain what I mean with some examples from this PR and the article you linked, but I don't want to hijack this thread without being asked.

**Comment by akien-mga** [comment]
> I recommend that you close this PR and create a new one with just the non-var-related changes.

Agreed.

**Pull request closed without merging**